lectures/tools_and_techniques/scalar_dynam.md

```diff
@@ -0,0 +1,356 @@
+
+k_{t+1} = s z k_t^{\alpha} + (1 - \delta) k_t
+```
+
+Here $k$ is capital stock and $s, z, \alpha, \delta$ are positive
+parameters with $0 < \alpha, \delta < 1$.
+
+If you try to iterate like we did in {eq}`sdslinmodpath`, you will find that
+the algebra gets messy quickly.
+
+Analyzing the dynamics of this model requires a different method (see below).
+
+### Stability
+
+A **steady state** of the difference equation $x_{t+1} = g(x_t)$ is a
+point $x^*$ in $S$ such that $x^* = g(x^*)$.
+
+In other words, $x^*$ is a **fixed point** of the function $g$ in
+$S$.
+
+For example, for the linear model $x_{t+1} = a x_t + b$, you can use the
+definition to check that
+
+* $x^* := b/(1-a)$ is a steady state whenever $a \not= 1$.
+* if $a = 1$ and $b=0$, then every $x \in \mathbb R$ is a
+  steady state.
+* if $a = 1$ and $b \not= 0$, then the linear model has no steady
+  state in $\mathbb R$.
+
+A steady state $x^*$ of $x_{t+1} = g(x_t)$ is called
+**globally stable** if, for all $x_0 \in S$,
+
+$$
+x_t = g^t(x_0) \to x^* \text{ as } t \to \infty
+$$
+
+For example, in the linear model $x_{t+1} = a x_t + b$ with $a
+\not= 1$, the steady state $x^*$
+
+* is globally stable if $|a| < 1$ and
+* fails to be globally stable otherwise.
+
+This follows directly from {eq}`sdslinmod`.
+
+A steady state $x^*$ of $x_{t+1} = g(x_t)$ is called
+**locally stable** if there exists an $\epsilon > 0$ such that
+
+$$
+| x_0 - x^* | < \epsilon
+\; \implies \;
+x_t = g^t(x_0) \to x^* \text{ as } t \to \infty
+$$
+
+Obviously every globally stable steady state is also locally stable.
+
+We will see examples below where the converse is not true.
+
+## Graphical Analysis
+
+As we saw above, analyzing the dynamics for nonlinear models is nontrivial.
+
+There is no single way to tackle all nonlinear models.
+
+However, there is one technique for one-dimensional models that provides a
+great deal of intuition.
+
+This is a graphical approach based on **45 degree diagrams**.
+
+Let's look at an example: the Solow model with dynamics given in {eq}`solow_lom2`.
+
+We begin with some plotting code that you can ignore at first reading.
+
+The function of the code is to produce 45 degree diagrams and time series
+plots.
+
+```{code-cell} julia
+function plot45(g, xmin, xmax, x0, num_arrows = 6)
+
+    xgrid = LinRange(xmin, xmax, 200)
+    
+    plt = plot(xgrid, xgrid, xlim = (xmin, xmax), ylim = (xmin, xmax), linecolor = :black, lw = 2)
+    plot!(plt, xgrid, g.(xgrid), linecolor = :blue, lw = 2)
+    
+    x = x0
+
+    arrow_iterator = 0:(num_arrows-1)
+
+    arrow_kwargs = (arrow = :closed, linecolor = :black, alpha = 0.5)
+    dash_kwargs = (linestyle = :dash, linecolor = :black, alpha = 0.5)
+
+    xticks = zeros(num_arrows) # Is there a better style than this?
+ 
+    for (i, j) in enumerate(arrow_iterator)
+
+        xticks[i] = x
+
+        if i == 1
+            plot!([x, x], [0, g.(x)]; arrow_kwargs...)
+            plot!([x, g.(x)], [g.(x), g.(x)]; arrow_kwargs...)
```

Comment:
Since you use the `g(x)` repeatedly in this section of the code do you want to assign a temporary above?  e.g. `x_p = g(x)` and then just refer to the `x_p` value throughout?  Or something like that?

Reply:
I prefer the `g(x)` here, since then I don't have to keep track of an additional temporary variable. However, I am open to change it if needed.